septentrion/cli.py

```diff
@@ -5,32 +5,54 @@
 """
 import logging
 import os
-from typing import Any, Dict
+from typing import Any, TextIO
 
 import click
 from click.types import StringParamType
 
-from septentrion import __version__, core, db, migrate, settings, style, utils
+from septentrion import (
+    __version__,
+    configuration,
+    core,
+    db,
+    exceptions,
+    migrate,
+    style,
+    utils,
+)
 
 logger = logging.getLogger(__name__)
 
 
-def get_context_settings() -> Dict[str, Any]:
+def load_config(ctx: click.Context, param: click.Parameter, value: TextIO) -> None:
+    if not value:
+        try:
+            file_contents, file = configuration.read_default_configuration_files()
+        except exceptions.NoDefaultConfiguration:
+            pass
+    else:
+        file = getattr(value, "name", "stdin")
+        logger.info(f"Reading configuration from {file}")
+        file_contents = value.read()
+
     try:
-        with open("septentrion.ini") as file:
-            config = file.read()
-    except FileNotFoundError:
-        config = ""
+        default = configuration.parse_configuration_file(file_contents)
+    except exceptions.NoSeptentrionSection:
+        if file in configuration.CONFIGURATION_FILES:
```

Comment:
@k4nar 
`if file in configuration.CONFIGURATION_FILES:`

so it won't fire up on setup.cfg. I'll make the variable names more explicit.